app/services/ccms/submitters/upload_documents_service.rb

```diff
@@ -17,31 +15,48 @@ def call # rubocop:disable Metrics/AbcSize
 
       private
 
+      def failed_upload_ids
+        @failed_upload_ids ||= submission.submission_documents.select { |document| document.status == 'failed' }&.map(&:id)&.join(', ')
+      end
+
       def upload_document(submission_document)
-        document_upload_requestor = CCMS::Requestors::DocumentUploadRequestor.new(submission.case_ccms_reference,
-                                                                                  submission_document.ccms_document_id,
-                                                                                  Base64.strict_encode64(pdf_binary(submission_document)),
-                                                                                  submission.legal_aid_application.provider.username)
-        tx_id = document_upload_requestor.transaction_request_id
-        response = document_upload_requestor.call
-        update_document_status(submission_document, tx_id, response)
+        document_upload_requestor = document_upload_requestor(submission_document)
+        response = update_document_status_and_return_response(submission_document, document_upload_requestor)
         submission_document.save!
         submission.save!
+        create_history('case_created', submission_document.status, document_upload_requestor.formatted_xml, response)
       rescue CcmsError => e
         submission_document.status = :failed
-        raise CcmsError, e
+        submission.save!
```

Comment:
How comes we submission.save! after an error is rescued ? 

If it's to save the submission no matter if there's an error, you could use 'ensure' 
e.g.
```
rescue CcmsError => e
...
ensure
submission.save!
```